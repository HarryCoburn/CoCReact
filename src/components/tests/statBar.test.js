import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import StatBar from "../statBar";

const mockStore = configureStore([]);

describe("StatBar Component", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      UI: {
        showStats: true
      },
      stats: {
        byID: {
          strength: {
            name: "Strength",
            value: 50,
            max: 100
          },
          toughness: {
            name: "Toughness",
            value: 75,
            max: 100
          },
          libido: {
            name: "Libido",
            value: 50,
            max: 100
          }
        },
        coreIDs: ["strength"],
        combatIDs: ["toughness"],
        advIDs: ["libido"]
      }
    });

    component = renderer.create(
      <Provider store={store}>
        <StatBar />
      </Provider>
    );
  });

  it("should match the snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
