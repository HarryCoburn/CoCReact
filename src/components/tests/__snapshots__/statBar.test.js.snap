// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatBar Component should match the snapshot 1`] = `
<div
  className="statBar visible"
>
  <h2>
    Testing
  </h2>
  <h2>
    Core Stats
  </h2>
  <div
    className="statFrame"
  >
    <div
      className="statGradient"
    >
      <p
        className="statLabel"
      >
        Strength
        : 
        50
         
      </p>
    </div>
  </div>
  <h2>
    Combat Stats
  </h2>
  <div
    className="statFrame"
  >
    <div
      className="statGradient"
    >
      <p
        className="statLabel"
      >
        Toughness
        : 
        75
         
      </p>
    </div>
  </div>
  <h2>
    Advancement
  </h2>
  <div
    className="statFrame"
  >
    <div
      className="statGradient"
    >
      <p
        className="statLabel"
      >
        Libido
        : 
        50
         
      </p>
    </div>
  </div>
  <div
    className="timeDisplay"
  >
    <p>
      Day#: 
      <br />
      Time: 
      :
    </p>
  </div>
  <div
    className="spriteDisplay"
  />
</div>
`;
